src/crates/mod.rs

```diff
@@ -39,6 +60,59 @@ impl Crate {
                 RustwideCrate::git(&format!("https://github.com/{}/{}", repo.org, repo.name))
             }
             Self::Local(name) => RustwideCrate::local(&LOCAL_CRATES_DIR.join(name)),
+            Self::Git(repo) => RustwideCrate::git(&repo.url),
+        }
+    }
+}
+
+impl TryFrom<&'_ PackageId> for Crate {
+    type Error = failure::Error;
+
+    fn try_from(pkgid: &PackageId) -> Fallible<Crate> {
+        let parts = &pkgid
+            .repr
+            .split_ascii_whitespace()
+            .flat_map(|s| {
+                // remove ()
+                s.trim_matches(|c: char| c.is_ascii_punctuation())
+                    // split resource and protocol
+                    .split('+')
+            })
+            .collect::<Vec<_>>();
+
+        match parts[..] {
+            [name, version, "registry", _] => Ok(Crate::Registry(RegistryCrate {
+                name: name.to_string(),
+                version: version.to_string(),
+            })),
+            [name, _, "path", _] => Ok(Crate::Local(name.to_string())),
```

Comment:
Path crates should not be considered local crates. Local crates are hardcoded to be in the `local-crates/` subdirectory of this repository, while path crates could point to another location in the filesystem.